packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue

```diff
@@ -0,0 +1,267 @@
+  canadaProvinces,
+  GDPR_COUNTRIES
+} from "../const/signup.const";
+
+const CANADA = "Canada";
+const USA = "United States";
+
+export default {
+  name: "TermsAndConditionsDialog",
+  components: {
+    SmartLink: () => import("./SmartLink"),
+    DialogWindow: () => import("./DialogWindow")
+  },
+  props: {
+    socialUrl: {
+      required: true,
+      type: String
+    }
+  },
+  data() {
+    return {
+      selectedCountry: "",
+      selectedRegion: "",
+      regionData: { type: "", list: [] }
+    };
+  },
+  computed: {
+    isDisabledSocialAuth() {
+      // If country not selected, disable "continue" button.
+      if (!this.selectedCountry) {
+        return true;
+      }
+      // If country Canada or USA but region not selected, disable "continue" button.
+      if (this.selectedCountry === USA || this.selectedCountry === CANADA) {
+        if (!this.selectedRegion) {
+          return true;
+        }
+      }
+    },
+    region: {
+      get() {
+        return this.regionData;
+      },
+      set(country) {
+        if (country === USA) {
+          this.regionData.list = americanStates;
+          this.regionData.type = "State";
+        } else if (country === CANADA) {
+          this.regionData.list = canadaProvinces;
+          this.regionData.type = "Province";
+        } else {
+          this.regionData.list = [];
+          this.regionData.type = "";
+        }
+      }
+    },
+    countrieList() {
+      return countriesList;
+    },
+    isGDPRCountry() {
+      if (GDPR_COUNTRIES.includes(this.selectedCountry)) {
+        return true;
+      }
+    }
+  },
+  methods: {
+    setTaCUrlAndRedirect() {
+      const acceptContactValue = this.$refs.gdprBoxEl.checked;
+      const countryName = this.selectedCountry;
+      const regionType = this.region.type.toLowerCase();
+      const regionName = this.selectedRegion;
+      const dataStamp = Date.now();
+      let redirectUrlParameters = [];
+
+      redirectUrlParameters = [
+        { name: "okta_AcceptedToS", value: acceptContactValue },
+        { name: "okta_ts_AcceptedToS", value: dataStamp },
+        { name: "country", value: encodeURI(countryName) },
+        { name: regionType, value: encodeURI(regionName) },
+      ];
+
+      window.location.href = this._getSocialRedirectUrl(redirectUrlParameters);
+    },
+    closeTermsConditionsDialog() {
+      this.$emit("close");
+    },
+    resetFields() {
+      if (this.$refs.gdprBoxEl) {
+        this.$refs.gdprBoxEl.checked = false;
+      }
+
+      if (this.$refs.regionDataEl) {
+        this.selectedRegion = "";
+        this.$refs.regionDataEl.value = "";
+      }
+    },
+    _getSocialRedirectUrl(parameters = []) {
+      let separator;
+      let redirectUrl = `${this.socialUrl}`;
+
+      parameters.forEach((parameter, index) => {
+        if (parameter.name) {
+          separator = index === 0 ? "?" : "&";
+          redirectUrl += `${separator}${parameter.name}=${parameter.value}`;
```

Comment:
Perhaps we should consider using a library or writing a small utility that does URI construction. This can get sticky fast.

Reply:
does it mean that we should create some helper class or mixin for URI purposes and to put this code inside?

Reply:
i've written this exact utility before but it can be tricky so using a library would be a good way to accomplish this too